src/sprint/managers/taskmanager/InMemoryTaskManager.java

```diff
@@ -185,36 +220,69 @@ private int generateTaskId() {
 
     @Override
     public List<SubTask> getSubTasksOfEpic(Epic epic) {
-        List<SubTask> subTasksList = new ArrayList<>();
-        if (epic != null) {
-            for (Integer subTaskId : epic.getSubTaskIds()) {
-                SubTask subTask = subTaskMap.get(subTaskId);
-                if (subTask != null) {
-                    subTasksList.add(subTask);
-                }
-            }
+        if (epic == null) {
+            return Collections.emptyList();
         }
-        return subTasksList;
+        return epic.getSubTaskIds().stream()
+                .map(subTaskMap::get)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
     }
 
+
     @Override
     public List<Task> getHistory() {
         return history.getHistory();
     }
 
-    protected void updateEpicStatusBySubTasks(Epic epic) {
-        boolean allSubTasksDone = true;
+
+    public boolean isCrossing(Task task) {
+        if (task.getStartTime() == null || task.getEndTime() == null) {
+            return false;
+        }
+
+        return getPrioritizedTasks().stream()
+                .filter(otherTask -> otherTask.getStartTime() != null && otherTask.getEndTime() != null)
+                .anyMatch(otherTask -> !(task.getEndTime().isBefore(otherTask.getStartTime()) ||
+                        otherTask.getEndTime().isBefore(task.getStartTime())));
+    }
+
+
+    protected void updateEpicBySubTasks(Epic epic) {
+        boolean allSubTasksDone = epic.getSubTaskIds().stream()
+                .map(subTaskId -> subTaskMap.get(subTaskId))
```

Comment:
Можно заменить на [method reference](https://www.baeldung.com/java-method-references), так код выглядит лаконичнее
```suggestion
                .map(subTaskMap::get)
```